lib/src/util/validation_util.dart

```diff
@@ -28,10 +28,16 @@ class ValidationUtil {
   /// Use this to log warnings to the console in dev mode only.
   /// This is to be used in assert calls for dev help only so that it gets
   /// compiled out for production.
+  ///
   ///     assert(ValidationUtil.warn('Some warning message'));
   ///
-  /// The message will get print out to the console.
-  static bool warn(String message) {
+  /// Optionally, a component can be passed as the second parameter
+  /// to provide additional information in the console.
+  ///
+  ///     assert(ValidationUtil.warn('Some warning message', component));
+  ///
+  /// The message will be printed out to the console.
+  static bool warn(String message, [dynamic component]) {
```

Comment:
Hm, or I wonder if we should accept `ReactElement`s as well (would we ever want to print just the child?)...

Could do something like:
```dart
static bool warn(String message, [dynamic data]) {
  ...

  if (data != null) {
    window.console.groupCollapsed('(Warning info)');
    window.console.log(data);

    if (isValidElement(data)) {
      window.console.log('props: ${prettyPrintMap(getProps(data))}');
    } else if (data is react.Component) {
      window.console.log('props: ${prettyPrintMap(data.props)}');
    }
  }

 ...
}